source/contribute.rst

```diff
@@ -192,6 +195,70 @@ this in case you want to read more about this:
 - `Commit message style for git <https://commit.style/>`_
 - `A note about git commit messages <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
 
+To squash or not to squash
+--------------------------
+
+Of course, if you want to craft good commit messages, you will have a
+hard time if the changeset you are describing does too many things. That
+might very well happen if you notice small things along the way that are
+unrelated to your PR, but too small to warrant a separate one. ``git add
+--patch`` or ``git add -p`` will be of invaluable help to commit things
+separately.
+On the contrary, there are commits that typically do not need to exist,
+such as commits that fix coding style or address minor review comments.
+Bear in mind that the git log is not only aimed at reviewers, but also
+at anyone who wants to understand some change you made. Do not distract
+them with cs fixes and try to make it look as if you produced perfect
+code from the start. Also, it's best if all of your commits pass the
+build, because that makes them ``git bisect`` friendly, but it also
+means they are likely to be revertable independently from other commits
+in your PR.
+The best way to avoid having to squash anything is to amend your last
+commit if that's indeed where your extra change is meant to go. That
+being said, sometimes you end up with many commits and it's too late for
+that.
+
+If you are in the fairly simple case where you want squash all your
+commits into one, you can take the following steps to achieve that:
+
+1. ``git rebase -i doctrine/3.2.x``, assuming ``doctrine`` is a git remote
+   that points to the target repository, and not your fork, and assuming
+   the target branch is 3.2.x. If you're not sure what your remotes are,
+   run ``git remote -vvv``, there should be your fork and the
+   holy/reference/base/doctrine/whatever-you-call-it repository.
+2. A window will show up with many lines, replace ``pick`` with
```

Comment:
Would it make more sense to point to some good tutorial on interactive rebasing? Maybe even instead listing the details here? Otherwise, if this info is not sufficient for a developer w/o the corresponding skills, the will be blocked.

Reply:
Maybe yeah, I think I have found something authoritative enough: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing
They use squash though, not fixup, so there is a risk that we end up with bad commit messages. Maybe let's experiment with that and see if it results in something like that or not…

Reply:
A new page could be created with more info about the different steps later if anything needs a deeper explanation. If the contribution entry page gets too big, it might discourage first time contributors.